src/tcpcrypt.c

```diff
@@ -1182,8 +1182,22 @@ static int do_output_pkconf_rcvd(struct tc *tc, struct ip *ip,
 	int len, klen;
 	struct tc_init1 *init1;
 	void *key;
+	struct tcpopt_eno *eno;
 	uint8_t *p;
 
+	/* Add the minimal ENO option to indicate support */
+	len = sizeof(*eno) + sizeof(tc->tc_cipher_pkey);
```

Comment:
We shouldn't be sending tc_cipher_pkey - just the minimal (empty) ENO option to indicate support